lib/layout/image_placeholder.dart

```diff
@@ -54,23 +54,48 @@ class FadeInImagePlaceholder extends StatelessWidget {
   /// See [Image.fit].
   final BoxFit fit;
 
+  @override
+  _FadeInImagePlaceholderState createState() => _FadeInImagePlaceholderState();
+}
+
+class _FadeInImagePlaceholderState extends State<FadeInImagePlaceholder> {
+  bool _isLoaded = false;
+
+  @override
+  void initState() {
+    super.initState();
+    // Once the image is loaded, we no longer need to show a placeholder
+    widget.image
+        .resolve(ImageConfiguration.empty)
+        .addListener(ImageStreamListener(
+      (imageInfo, synchronousCall) {
+        setState(() {
+          _isLoaded = true;
+        });
+      },
+    ));
+  }
+
   @override
   Widget build(BuildContext context) {
     return Image(
-      image: image,
-      excludeFromSemantics: excludeFromSemantics,
-      width: width,
-      height: height,
-      fit: fit,
+      image: widget.image,
+      excludeFromSemantics: widget.excludeFromSemantics,
+      width: widget.width,
+      height: widget.height,
+      fit: widget.fit,
       frameBuilder: (context, child, frame, wasSynchronouslyLoaded) {
         // We only need to animate the loading of images on the web. We can
         // also return early if the images are already in the cache.
         if (wasSynchronouslyLoaded || !kIsWeb) {
-          return this.child ?? child;
+          if (!_isLoaded) {
```

Comment:
Why do we need the extra _isLoaded flag? How does it differ from wasSynchronouslyLoaded?